core/embed/trezorhal/random_delays.c

```diff
@@ -34,101 +34,92 @@ Random Delays in Embedded Software by Michael Tunstall, Olivier Benoit:
 https://link.springer.com/content/pdf/10.1007%2F978-3-540-72354-7_3.pdf
 */
 
-#include "rdi.h"
+#include "random_delays.h"
 
 #include <stdbool.h>
 
 #include "chacha_drbg.h"
 #include "common.h"
+#include "drbg.h"
 #include "memzero.h"
 #include "rand.h"
-#include "secbool.h"
+
+// from util.s
+extern void shutdown(void);
 
 #define BUFFER_LENGTH 64
-#define RESEED_INTERVAL 65536
 
-static CHACHA_DRBG_CTX drbg_ctx;
 static uint8_t buffer[BUFFER_LENGTH];
-static size_t buffer_index;
+static size_t buffer_index = 0;
```

Comment:
I don't know what you mean exactly since `buffer` is used in both `random8` and `buffer_refill` but I inlined `buffer_refill` and made both variables local. See https://github.com/trezor/trezor-firmware/commit/00e71ec437943a725a8c3f5dec97565936843769.